docs/architecture/transaction.md

```diff
@@ -0,0 +1,64 @@
+# Transaction
+
+> Single account state transition
+
+## What is the purpose of a transaction?
+
+To facilitate single account state transitions.
+
+## What is a transaction?
+
+In Miden, a `Transaction` represents the state transition of a single account. A `Transaction` takes a single [account](accounts.md) and one or more [notes](notes.md), and none or one script (piece of code executed after all notes have been executed) as input, and outputs the same account with a potentially updated state, together with some potential newly created notes.
+
+![Transaction diagram](../img/architecture/transaction/transaction-diagram.png)
+
+## Transaction core components
+
+WIP
+
+## Transaction architecture
+
+Miden aims for the following characteristics in `Transaction`s:
+
+- **Parallel transaction execution**: Because a transaction is always performed against a single account, Miden obtains asynchronicity. 
+- **Private transaction execution**: Local execution of transactions enables preservation of sensitive data.
+
+There are two types of transactions in Miden: **local transactions** and **network transactions**.
+
+![Local vs network transactions](../img/architecture/transaction/local-vs-network-transaction.png)
+
+### Local transactions
+
+This is where clients executing the transactions also generate the proofs of their correct execution. So, no additional work needs to be performed by the network.
+
+Local transactions are useful for several reasons:
+
+1. They are cheaper (i.e., lower fees) as zk-proofs are already generated by the clients.
+2. They allow fairly complex computations because the proof size doesn't grow linearly with the complexity of the computation.
+3. They enable privacy as neither the account state nor account code are needed to verify the zk-proof.
```

Comment:
I think we can improve this section to make sure it really stands out. Local transactions is something that most other chains cannot do. They basically enabled theoretically unbounded computation complexity (e.g., no more gas limits). I would also make sure that this is not just for users - apps can benefit from this too. e.g., we can have apps running their logic off-chain (either in a centralized or decentralized mode) to minimize fees, not be subject to network congestion etc.